src/compiler/c.rs

```diff
@@ -202,17 +206,14 @@ impl<T, I> CompilerHasher<T> for CCompilerHasher<I>
             }
         }).and_then(move |preprocessor_result| {
             trace!("[{}]: Preprocessor output is {} bytes",
-                   parsed_args.output_file(),
+                   parsed_args.output_pretty(),
                    preprocessor_result.stdout.len());
 
-            // Remove object file from arguments before hash calculation
             let key = {
-                let out_file = parsed_args.output_file();
-                let arguments = parsed_args.common_args.iter()
-                    .filter(|a| **a != out_file)
-                    .map(|a| a.as_str())
-                    .collect::<String>();
-                hash_key(&executable_digest, &arguments, &env_vars, &preprocessor_result.stdout)
+                hash_key(&executable_digest,
+                         &parsed_args.common_args,
```

Comment:
Oh right sorry I meant to ask you about this. I wasn't really sure if this was still necessary. The test here is against `output_file` which is what prompted me to rename the relevant method to `output_pretty` because the return value of `output_file` is just the file name, not the full path.

In that sense I'm not sure that this was 100% working before, and I otherwise couldn't figure out what it was intended to be doing. I thought the output file was already missing from the `common_args`?